ui/webui/test/machine_install.py

```diff
@@ -62,18 +62,44 @@ def _create_disk_image(self, size, image_path=None, quiet=False):
         self._execute(f"qemu-img create -f qcow2 {quiet} {image_path} {size}G")
         return image_path
 
+    def _serve_updates_img(self):
+        http_port = self._get_free_port()
+        self.http_server = subprocess.Popen(["python3", "-m", "http.server", "-d", ANACONDA_ROOT_DIR, str(http_port)])
+        self.http_server.communicate()
```

Comment:
This is the problem with the forever-hanging test. communicate() waits until the spawned process exits, which will never happen until you kill it. Instead, use `m.wait_for_cockpit_running('127.0.0.1', http_port)` until it is up, let it run in the background, and set up a cleanup handler which does `.terminate()` and `.wait()`.